From d11e60203e74d8b5e5a35764804ed8b9b7775c89 Mon Sep 17 00:00:00 2001
From: atkac <atkac@3789f03b-4d11-0410-bbf8-ca57d06f2519>
Date: Tue, 18 May 2010 13:35:37 +0000
Subject: [PATCH] [Bugfix] Use always XkbConvertCase instead of XConvertCase when compiling
 against X.Org 1.8.X and newer. Othewrise keysyms are not mapped correctly
 to keycodes.

Reference: https://bugzilla.redhat.com/show_bug.cgi?id=586406


git-svn-id: https://tigervnc.svn.sourceforge.net/svnroot/tigervnc/trunk@4057 3789f03b-4d11-0410-bbf8-ca57d06f2519
---
 unix/xserver/hw/vnc/Input.cc |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/unix/xserver/hw/vnc/Input.cc b/unix/xserver/hw/vnc/Input.cc
index d108763..f1e4801 100644
--- a/unix/xserver/hw/vnc/Input.cc
+++ b/unix/xserver/hw/vnc/Input.cc
@@ -712,7 +712,12 @@ static KeySym KeyCodetoKeySym(KeySymsPtr keymap, int keycode, int col)
 	}
 
 	if ((per <= (col|1)) || (syms[col|1] == NoSymbol)) {
-		XConvertCase(syms[col&~1], &lsym, &usym);
+#if XORG >= 18
+		XkbConvertCase
+#else
+		XConvertCase
+#endif
+			    (syms[col&~1], &lsym, &usym);
 		if (!(col & 1))
 			return lsym;
 		/*
-- 
1.5.6.5

